events/tests/importers/test_espoo.py

```diff
@@ -0,0 +1,367 @@
+        super_event_type=Event.SuperEventType.UMBRELLA,
+        publisher=common_org.id,
+    )
+    event_jsonld = {"@id": f"https://localhost/{event_data['id']}/"}
+    recurring_event1_data = EventDataFactory(
+        super_event=event_jsonld,
+        super_event_type=Event.SuperEventType.RECURRING,
+        publisher=common_org.id,
+    )
+    recurring_event2_data = EventDataFactory(
+        super_event=event_jsonld,
+        super_event_type=Event.SuperEventType.RECURRING,
+        publisher=common_org.id,
+    )
+
+    # Setup data into non-hierarchical order
+    origin_data = [
+        recurring_event1_data,
+        super_event_data,
+        event_data,
+        recurring_event2_data,
+    ]
+
+    importer_ds = DataSourceFactory(id="importer_ds")
+    instance_map, synch = _import_origin_objs(
+        Event,
+        importer_ds,
+        [],
+        origin_data,
+        copy_fields=["super_event_type"],
+        pre_field_mappers={
+            "publisher": _pre_map_to_id({common_org.id: common_org.id}),
+        },
+        post_field_mappers={
+            "super_event": _post_map_self_id(True),
+        },
+    )
+    synch.finish()
+
+    assert len(instance_map) == 4
+
+    super_event = Event.objects.get(origin_id=super_event_data["id"])
+    event = Event.objects.get(origin_id=event_data["id"])
+    recurring_event1 = Event.objects.get(origin_id=recurring_event1_data["id"])
+    recurring_event2 = Event.objects.get(origin_id=recurring_event2_data["id"])
+
+    assert super_event.super_event is None
+    assert event.super_event == super_event
+    assert recurring_event1.super_event == event
+    assert recurring_event2.super_event == event
+
+    assert super_event.super_event_type is None
+    assert event.super_event_type == Event.SuperEventType.UMBRELLA
+    assert recurring_event1.super_event_type == Event.SuperEventType.RECURRING
+    assert recurring_event2.super_event_type == Event.SuperEventType.RECURRING
+
+
+@pytest.mark.django_db
+def test_split_common_objs():
+    common_ds = DataSourceFactory(id="common_ds")
+    importer_ds = DataSourceFactory(id="importer_ds")
+    common_org = OrganizationFactory(data_source=common_ds)
+    imported_org = OrganizationFactory(
+        data_source=importer_ds, origin_id="imported_org_id"
+    )
+
+    origin_data = [
+        {"id": common_org.id},
+        {"id": imported_org.origin_id, "data_source": "origin_ds"},
+        {"id": "new_org_id", "data_source": "origin_ds"},
+    ]
+
+    common_objs, origin_objs = _split_common_objs(
+        Organization, importer_ds, origin_data
+    )
+    assert [d["id"] for d in origin_data[:1]] == [i.id for i in common_objs]
+    assert origin_data[1:] == origin_objs
+
+
+@pytest.mark.django_db
+def test_importer(settings, requests_mock, sleep, api_client):
+    settings.ESPOO_API_URL = "http://localhost/"
+    settings.ESPOO_API_QUERY = "test=1"
+
+    # Generate the test data in our own API
```

Comment:
It's a good idea to utilize our API for generating the mock responses. However, it's not a good idea to make this dynamic; while Espoo uses Linked Events, it's their own version of Linked Events and is not guaranteed to match with ours. Use a fixture instead.